io/ringbuffer.go

```diff
@@ -70,3 +70,9 @@ func (r *RingBuffer) Read(p []byte) (int, error) {
 	}
 	return readCount, nil
 }
+
+// Reset resets the ring buffer.
+func (r *RingBuffer) Reset() {
+	r.slice = r.slice[0:]
+	r.start, r.end, r.size = 0, 0, 0
+}
```

Comment:
would be nice to have a unit test using this method.

Reply:
Updated with test. Thanks!